cmd/argoexec/commands/root.go

```diff
@@ -72,6 +74,7 @@ func NewRootCommand() *cobra.Command {
 func initExecutor() *executor.WorkflowExecutor {
 	log.WithField("version", argo.GetVersion().Version).Info("Starting Workflow Executor")
 	config, err := clientConfig.ClientConfig()
+	config = restclient.AddUserAgent(config, fmt.Sprintf("argo-workflow-%s-executor", os.Getenv(common.EnvVarContainerRuntimeExecutor)))
```

Comment:
user agent should probably be something like
```
User-Agent: <product> / <product-version> <comment>
```

E.g.

```
argo-workflows/v3.0.0 executor/docker
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Reply:
Thanks! Just updated the UA to follow the convention.